### PR TITLE
Warcraft dapp

### PR DESCRIPTION
**Pull request by amityadav0**

### CODE REVIEW

File: warcraft/dapp/src/components/indexer.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
+const algosdk = require('algosdk');
+
+const { getDefaultAccountAddr } = require('./payment-widget');
+const { indexerToken, indexerServer, indexerPort, assetIndex } = require('../config');
+
+const indexerClient = new algosdk.Indexer(indexerToken, indexerServer, indexerPort);
+
+// after_time function in indexer accepts RFC 3339 date-time format
+// https://stackoverflow.com/questions/7244246/generate-an-rfc-3339-timestamp-similar-to-google-tasks-api
+function ISODateString(d){
```
Comment (ratik21) [inline warcraft/dapp/src/components/indexer.js]: Can we rename `ISODateString` to `toISODateString`?

File: warcraft/dapp/src/components/indexer.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+// Detect if user made weekly payment.(returns true if user did)
+export async function detectPayment() {
+  const address = await getDefaultAccountAddr();
+  // time before 7 days
+  let beforeTime = new Date();
+  beforeTime.setDate(beforeTime.getDate() - 7);
+
+  let response = await indexerClient.searchForTransactions().address(address).afterTime(ISODateString(beforeTime))
+                  .txType('axfer').assetID(assetIndex).currencyGreaterThan(0).do();
```
Comment (ratik21) [inline warcraft/dapp/src/components/indexer.js]: I think this searches for payments "from" & "to" this `address`. We only want to track receiving payments, right? Could you please check/validate?

File: warcraft/dapp/src/components/indexer.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+  console.log(response);
+  if(response.transactions.length > 0) {
+      return true;
+  }
+  return false;
```
Comment (ratik21) [inline warcraft/dapp/src/components/indexer.js]:
```suggestion
  return response.transactions.length > 0;
```

**Review by ratik21 — COMMENTED** [review]
Please update the readme, regarding the added functionality in warcraft dapp

File: warcraft/dapp/src/components/indexer.js (hunk 1 above)
Comment (amityadav0) [reply warcraft/dapp/src/components/indexer.js]: yes

File: warcraft/dapp/src/components/indexer.js (hunk 2 above)
Comment (amityadav0) [reply warcraft/dapp/src/components/indexer.js]: checking

Comment (amityadav0) [reply warcraft/dapp/src/components/indexer.js]: fixed

**Comment by KenG-pro** [comment]
Patch

**Review by robert-zaremba — APPROVED** [review]

**Pull request merged**